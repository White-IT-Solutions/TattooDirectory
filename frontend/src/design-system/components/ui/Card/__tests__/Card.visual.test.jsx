import { render } from '@testing-library/react';
import { 
  Card, 
  CardHeader, 
  CardTitle, 
  CardDescription, 
  CardContent, 
  CardFooter 
} from '../Card';

// Mock for visual regression testing
// In a real implementation, this would use tools like Chromatic, Percy, or jest-image-snapshot
describe('Card Visual Regression Tests', () => {
  describe('Elevation Variants', () => {
    it('flat elevation should match snapshot', () => {
      const { container } = render(
        <Card elevation="flat" className="w-64">
          <CardContent>Flat elevation card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-flat-elevation');
    });

    it('low elevation should match snapshot', () => {
      const { container } = render(
        <Card elevation="low" className="w-64">
          <CardContent>Low elevation card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-low-elevation');
    });

    it('medium elevation should match snapshot', () => {
      const { container } = render(
        <Card elevation="medium" className="w-64">
          <CardContent>Medium elevation card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-medium-elevation');
    });

    it('high elevation should match snapshot', () => {
      const { container } = render(
        <Card elevation="high" className="w-64">
          <CardContent>High elevation card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-high-elevation');
    });

    it('floating elevation should match snapshot', () => {
      const { container } = render(
        <Card elevation="floating" className="w-64">
          <CardContent>Floating elevation card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-floating-elevation');
    });
  });

  describe('Padding Variants', () => {
    it('no padding should match snapshot', () => {
      const { container } = render(
        <Card padding="none" className="w-64">
          <CardContent>No padding card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-no-padding');
    });

    it('small padding should match snapshot', () => {
      const { container } = render(
        <Card padding="sm" className="w-64">
          <CardContent>Small padding card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-small-padding');
    });

    it('medium padding should match snapshot', () => {
      const { container } = render(
        <Card padding="md" className="w-64">
          <CardContent>Medium padding card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-medium-padding');
    });

    it('large padding should match snapshot', () => {
      const { container } = render(
        <Card padding="lg" className="w-64">
          <CardContent>Large padding card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-large-padding');
    });
  });

  describe('Radius Variants', () => {
    it('no radius should match snapshot', () => {
      const { container } = render(
        <Card radius="none" className="w-64">
          <CardContent>No radius card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-no-radius');
    });

    it('small radius should match snapshot', () => {
      const { container } = render(
        <Card radius="sm" className="w-64">
          <CardContent>Small radius card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-small-radius');
    });

    it('large radius should match snapshot', () => {
      const { container } = render(
        <Card radius="lg" className="w-64">
          <CardContent>Large radius card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-large-radius');
    });

    it('extra large radius should match snapshot', () => {
      const { container } = render(
        <Card radius="xl" className="w-64">
          <CardContent>Extra large radius card</CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-xl-radius');
    });
  });

  describe('Complete Card Compositions', () => {
    it('basic card with header and content should match snapshot', () => {
      const { container } = render(
        <Card className="w-80">
          <CardHeader>
            <CardTitle>Card Title</CardTitle>
            <CardDescription>This is a card description that explains the content.</CardDescription>
          </CardHeader>
          <CardContent>
            <p>This is the main content of the card. It can contain any type of content.</p>
          </CardContent>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-basic-composition');
    });

    it('complete card with all components should match snapshot', () => {
      const { container } = render(
        <Card className="w-80">
          <CardHeader>
            <CardTitle>Complete Card</CardTitle>
            <CardDescription>A card with all composition components.</CardDescription>
          </CardHeader>
          <CardContent>
            <div className="space-y-2">
              <p>Main content area with multiple elements.</p>
              <div className="bg-gray-100 p-2 rounded">
                <span className="text-sm">Nested content</span>
              </div>
            </div>
          </CardContent>
          <CardFooter>
            <button className="px-4 py-2 bg-blue-500 text-white rounded">
              Action Button
            </button>
          </CardFooter>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot('card-complete-composition');
    });

    it('card with different elevation and padding combinations should match snapshot', () => {
      const { container } = render(
        <div className="space-y-4">
          <Card elevation="flat" padding="sm" className="w-64">
            <CardContent>Flat + Small Padding</CardContent>
          </Card>
          <Card elevation="high" padding="lg" className="w-64">
            <CardContent>High + Large Padding</CardContent>
          </Card>
          <Card elevation="floating" padding="md" className="w-64">
            <CardContent>Floating + Medium Padding</CardContent>
          </Card>
        </div>
      );
      expect(container).toMatchSnapshot('card-elevation-padding-combinations');
    });

    it('responsive card layout should match snapshot', () => {
      const { container } = render(
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
          <Card>
            <CardHeader>
              <CardTitle>Card 1</CardTitle>
            </CardHeader>
            <CardContent>Responsive grid layout</CardContent>
          </Card>
          <Card>
            <CardHeader>
              <CardTitle>Card 2</CardTitle>
            </CardHeader>
            <CardContent>Responsive grid layout</CardContent>
          </Card>
          <Card>
            <CardHeader>
              <CardTitle>Card 3</CardTitle>
            </CardHeader>
            <CardContent>Responsive grid layout</CardContent>
          </Card>
        </div>
      );
      expect(container).toMatchSnapshot('card-responsive-layout');
    });
  });

  describe('Interactive States', () => {
    it('card with hover effects should match snapshot', () => {
      const { container } = render(
        <div className="space-y-4">
          <Card elevation="low" className="w-64 hover:shadow-md">
            <CardContent>Hover for shadow change</CardContent>
          </Card>
          <Card elevation="floating" className="w-64">
            <CardContent>Hover for lift effect</CardContent>
          </Card>
        </div>
      );
      expect(container).toMatchSnapshot('card-hover-states');
    });
  });

  describe('Dark Mode Compatibility', () => {
    it('card in dark mode should match snapshot', () => {
      const { container } = render(
        <div className="dark">
          <Card className="w-80">
            <CardHeader>
              <CardTitle>Dark Mode Card</CardTitle>
              <CardDescription>This card should work in dark mode.</CardDescription>
            </CardHeader>
            <CardContent>
              <p>Content that adapts to dark mode colors.</p>
            </CardContent>
            <CardFooter>
              <button className="px-4 py-2 bg-primary-500 text-white rounded">
                Dark Mode Button
              </button>
            </CardFooter>
          </Card>
        </div>
      );
      expect(container).toMatchSnapshot('card-dark-mode');
    });
  });
});