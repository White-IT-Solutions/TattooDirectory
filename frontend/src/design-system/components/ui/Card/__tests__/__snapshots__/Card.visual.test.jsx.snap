// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Visual Regression Tests Complete Card Compositions basic card with header and content should match snapshot: card-basic-composition 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-80"
>
  <div
    class="flex flex-col space-y-1.5 "
  >
    <h3
      class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
    >
      Card Title
    </h3>
    <p
      class="text-sm text-[var(--text-secondary)] "
    >
      This is a card description that explains the content.
    </p>
  </div>
  <div
    class=""
  >
    <p>
      This is the main content of the card. It can contain any type of content.
    </p>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Complete Card Compositions card with different elevation and padding combinations should match snapshot: card-elevation-padding-combinations 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-none border-[var(--border-primary)] p-4 rounded-[var(--radius-lg)] w-64"
    >
      <div
        class=""
      >
        Flat + Small Padding
      </div>
    </div>
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-lg)] hover:shadow-[var(--shadow-xl)] border-transparent p-8 rounded-[var(--radius-lg)] w-64"
    >
      <div
        class=""
      >
        High + Large Padding
      </div>
    </div>
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-xl)] hover:shadow-[var(--shadow-2xl)] border-transparent hover:-translate-y-1 p-6 rounded-[var(--radius-lg)] w-64"
    >
      <div
        class=""
      >
        Floating + Medium Padding
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Complete Card Compositions complete card with all components should match snapshot: card-complete-composition 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-80"
>
  <div
    class="flex flex-col space-y-1.5 "
  >
    <h3
      class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
    >
      Complete Card
    </h3>
    <p
      class="text-sm text-[var(--text-secondary)] "
    >
      A card with all composition components.
    </p>
  </div>
  <div
    class=""
  >
    <div
      class="space-y-2"
    >
      <p>
        Main content area with multiple elements.
      </p>
      <div
        class="bg-gray-100 p-2 rounded"
      >
        <span
          class="text-sm"
        >
          Nested content
        </span>
      </div>
    </div>
  </div>
  <div
    class="flex items-center pt-4 "
  >
    <button
      class="px-4 py-2 bg-blue-500 text-white rounded"
    >
      Action Button
    </button>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Complete Card Compositions responsive card layout should match snapshot: card-responsive-layout 1`] = `
<div>
  <div
    class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4"
  >
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)]"
    >
      <div
        class="flex flex-col space-y-1.5 "
      >
        <h3
          class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
        >
          Card 1
        </h3>
      </div>
      <div
        class=""
      >
        Responsive grid layout
      </div>
    </div>
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)]"
    >
      <div
        class="flex flex-col space-y-1.5 "
      >
        <h3
          class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
        >
          Card 2
        </h3>
      </div>
      <div
        class=""
      >
        Responsive grid layout
      </div>
    </div>
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)]"
    >
      <div
        class="flex flex-col space-y-1.5 "
      >
        <h3
          class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
        >
          Card 3
        </h3>
      </div>
      <div
        class=""
      >
        Responsive grid layout
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Dark Mode Compatibility card in dark mode should match snapshot: card-dark-mode 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-80"
    >
      <div
        class="flex flex-col space-y-1.5 "
      >
        <h3
          class="text-lg font-semibold leading-none tracking-tight text-[var(--text-primary)] "
        >
          Dark Mode Card
        </h3>
        <p
          class="text-sm text-[var(--text-secondary)] "
        >
          This card should work in dark mode.
        </p>
      </div>
      <div
        class=""
      >
        <p>
          Content that adapts to dark mode colors.
        </p>
      </div>
      <div
        class="flex items-center pt-4 "
      >
        <button
          class="px-4 py-2 bg-primary-500 text-white rounded"
        >
          Dark Mode Button
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Elevation Variants flat elevation should match snapshot: card-flat-elevation 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-none border-[var(--border-primary)] p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Flat elevation card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Elevation Variants floating elevation should match snapshot: card-floating-elevation 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-xl)] hover:shadow-[var(--shadow-2xl)] border-transparent hover:-translate-y-1 p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Floating elevation card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Elevation Variants high elevation should match snapshot: card-high-elevation 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-lg)] hover:shadow-[var(--shadow-xl)] border-transparent p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    High elevation card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Elevation Variants low elevation should match snapshot: card-low-elevation 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-sm)] hover:shadow-[var(--shadow)] border-[var(--border-secondary)] p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Low elevation card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Elevation Variants medium elevation should match snapshot: card-medium-elevation 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Medium elevation card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Interactive States card with hover effects should match snapshot: card-hover-states 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-sm)] hover:shadow-[var(--shadow)] border-[var(--border-secondary)] p-6 rounded-[var(--radius-lg)] w-64 hover:shadow-md"
    >
      <div
        class=""
      >
        Hover for shadow change
      </div>
    </div>
    <div
      class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow-xl)] hover:shadow-[var(--shadow-2xl)] border-transparent hover:-translate-y-1 p-6 rounded-[var(--radius-lg)] w-64"
    >
      <div
        class=""
      >
        Hover for lift effect
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Visual Regression Tests Padding Variants large padding should match snapshot: card-large-padding 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-8 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Large padding card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Padding Variants medium padding should match snapshot: card-medium-padding 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Medium padding card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Padding Variants no padding should match snapshot: card-no-padding 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-0 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    No padding card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Padding Variants small padding should match snapshot: card-small-padding 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-4 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Small padding card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Radius Variants extra large radius should match snapshot: card-xl-radius 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-xl)] w-64"
>
  <div
    class=""
  >
    Extra large radius card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Radius Variants large radius should match snapshot: card-large-radius 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius-lg)] w-64"
>
  <div
    class=""
  >
    Large radius card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Radius Variants no radius should match snapshot: card-no-radius 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-none w-64"
>
  <div
    class=""
  >
    No radius card
  </div>
</div>
`;

exports[`Card Visual Regression Tests Radius Variants small radius should match snapshot: card-small-radius 1`] = `
<div
  class="bg-[var(--background-primary)] border border-[var(--border-subtle)] transition-all duration-200 ease-out overflow-hidden shadow-[var(--shadow)] hover:shadow-[var(--shadow-md)] border-[var(--border-subtle)] p-6 rounded-[var(--radius)] w-64"
>
  <div
    class=""
  >
    Small radius card
  </div>
</div>
`;
